via/services/google_drive.py

```diff
@@ -183,17 +179,55 @@ def iter_file(self, file_id, resource_key=None) -> Iterator[ByteString]:
                     resource_key,
                 )
 
+        headers = {}
         if resource_key:
             headers["X-Goog-Drive-Resource-Keys"] = f"{file_id}/{resource_key}"
 
         response = self._session.get(
```

Comment:
I tried to use the new service's iter_url here but the google part:
- Uses a session which looks like a request.session but is not. Taking a session on the new service works but I think it's confusing.
- Uses `@iter_handle_errors(translate_google_error)` for error handling. Using something similar in the new service will shallow the exceptions so the translation here wound't work. I could refactor the error handling on both to make it work but I think what I tried was too complicated at the end.